### ISSUE TITLE
 Session Manager plugin local install fails on ARM64 (M1) Mac with 'bad CPU type in executable'

### ISSUE THREAD

**Issue by gidsg** [aws/copilot-cli#4891]
## Steps to reproduce
1. have a copilot app deployed
1. On M1 Mac run: `copilot svc exec`
1. follow prompt to install Session Manager: `Looks like the Session Manager plugin is not installed yet.
Would you like to install the plugin to execute into the container? Yes`

## Actual Results
`✘ execute command /bin/sh in container app-name: start session ecs-execute-command-0cf70aeea8692d975 using ssm plugin: start session: fork/exec /usr/local/bin/session-manager-plugin: bad CPU type in executable`

## Expected Results
Command is successful 
Installs M1 version as per https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html#install-plugin-macos

**Comment by huanjani** [comment]
Hello, @gidsg!

Thanks for opening this issue. It looks like this may be related to https://github.com/aws/aws-cli/issues/7214. Let me see if there's a workaround....

**Comment by huanjani** [comment]
I'm seeing a lot of suggestions to install Rosetta 2...is that something you are willing to do?

**Comment by huanjani** [comment]
We will work on adding the Mac ARM file to Copilot; Rosetta 2 will get you running in the meantime.

**Comment by gidsg** [comment]
@huanjani to let you know I found a workaround which does not require Rosetta 2. Now there is a native binary, you need to manually install the ARM version outside of copilot, as per: https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html#install-plugin-macos
i.e.
```
curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/mac_arm64/sessionmanager-bundle.zip" -o "sessionmanager-bundle.zip"
unzip sessionmanager-bundle.zip
sudo ./sessionmanager-bundle/install -i /usr/local/sessionmanagerplugin -b /usr/local/bin/session-manager-plugin
```

I believe to fix this in copilot it would need to switch based on the system architecture, and change the downloaded file path, i.e. as currently defined in:
https://github.com/aws/copilot-cli/blob/cb542b8a67792b57de3281d99162ad5a7db23800/internal/pkg/exec/ssm_plugin_install_latest_binary_darwin.go#L14

**Comment by KollaAdithya** [comment]
Fix is now released in v1.28 https://github.com/aws/copilot-cli/releases/tag/v1.28.0. You can install the SSM plugin binary for ARM64 (M1) Mac. Thanks @gidsg  for reporting this!
